pkg/helm/manifest.go

```diff
@@ -63,6 +82,45 @@ func (o *ManifestObjectFilter) Matches(obj *resource.Info) bool {
 	return true
 }
 
+// ObjectMapFromReleaseWithMultipleFilters returns a map of objects from the helm release manifest matching the filter or an error.
+func ObjectMapFromReleaseWithMultipleFilters(restClientGetter genericclioptions.RESTClientGetter, r *release.Release, f *ManifestMultipleObjectFilter) ([]ObjectList, error) {
+	return ObjectMapFromManifestWithMultipleFilters(restClientGetter, r.Namespace, r.Manifest, f)
+}
+
+// ObjectMapFromManifestWithMultipleFilters returns a map of objects from the manifests matching the filter or an error.
+func ObjectMapFromManifestWithMultipleFilters(restClientGetter genericclioptions.RESTClientGetter, namespace, manifest string, f *ManifestMultipleObjectFilter) ([]ObjectList, error) {
+	allObjects := ObjectList{}
+	filteredObjects := []ObjectList{}
+	r, err := loadManifest(restClientGetter, namespace, manifest)
+	if err != nil {
+		return nil, fmt.Errorf("error loading manifest: %w", err)
+	}
+	err = r.Visit(func(info *resource.Info, err error) error {
+		if err != nil {
+			return err
+		}
+		if f != nil {
+			for _, filter := range f.Filters {
+				if filter.Matches(info) {
```

Comment:
This will execute the same filters n Times. It should be fine to run `f.Matches` once.
```golang
func (f *ManifestMultipleObjectFilter) Matches(obj *resource.Info) bool {
	for _, filter := range f.Filters {
		if filter.Matches(obj) {
			return true
		}
	}
	return false
}```